galley/pkg/config/analysis/msg/messages.yaml

```diff
@@ -177,4 +177,26 @@ messages:
       - name: portName
         type: string
       - name: host
-        type: string
\ No newline at end of file
+        type: string
+
+  - name: "DeploymentAssociatedToMultipleServices"
+    code: IST0115
+    level: Warning
+    description: "The resulting pods of a Deployment can't be associated to multiple Services in the same port but different protocol"
+    template: "The following services are associated to deployment %s at port %d but different protocol: %v."
+    args:
+      - name: deployment
+        type: string
+      - name: port
+        type: int32
+      - name: services
+        type: "[]string"
+
+  - name: "DeploymentRequiresServiceAssociated"
+    code: IST0116
+    level: Warning
+    description: "The resulting pods of a Deployment must be associated to at least one service"
```

Comment:
Approached this. In the latest commit, only the deployments not in the mesh (`sidecar.istio.io/inject: "false"` or namespace with disabled autoinjection) are analyzed. Message changed also. Do not hesitate to suggest another message for that.

Reply:
My suggestions:

**DeploymentAssociatedToMultipleServices.description:**

`The resulting pods of a service mesh deployment can't be associated with multiple services using the same port but different protocols`

**DeploymentAssociatedToMultipleServices.template:**

`This deployment is associated with multiple services using port %d but different protocols: %v`

Including the deployment as an arg is redundant, since it's the origin resource that the message is associated with.

**DeploymentRequiresServiceAssociated.description:**

`The resulting pods of a service mesh deployment must be associated with at least one service`

**DeploymentRequiresServiceAssociated.template:**

`No service associated with this deployment. Service mesh deployments must be associated with a service.`

Again, no need to include the deployment name since that's already covered.